Fixed #589 (by adding skipObject() and skipArray() methods)

greek1979 opened the pull request:
A fix for a straightforward issue #589 discovered by @DXTR66. The fix is about letting ``JsonStructureToParserAdapter `` parser adapter implement ``skipObject()`` and ``skipArray()`` methods, so both can be used by the caller without a fear of raising an unsupported-method-exception.

greek1979 commented:
After another, related issue was brought up on Friday, had to revisit the original patch, debug new issue extensively, and find a better solution that resolves both the original unmapped-property-in-polymorphic-object and multiple-nested-polymorphic-objects bugs.

The root cause for the latter was the same ``JsonStructureToParserAdapter`` parser adapter that was violating the contract for ``jakarta.json.stream.JsonParser.getObject()`` method, which is "Returns a JsonObject and advances the parser to thecorresponding END_OBJECT". But the adapter did not advance the internal JSON object parser to the end of current object, so after the caller was done with deserialization of nested (child) object, the adapter knew nothing about it, and attempted to proceed with existing key-iterator like nothing happened. Triggering a deserialization error, of course.

remimarechal commented:
This PR fixes my issues #592

Verdent commented:
Please, rebase on the latest master. I made fixes for copyright issues you are encountering here. (You might need to change some of the copyrights yourself, but only for files, you have changed) :-)

greek1979 commented:
@Verdent , thank you for pointing this out. Updated copyright year to 2023 for the files I have touched with my commits, rebased, and pushed to the feature branch. Should be ok now.

redmitry commented:
sorry for duplicating:
https://github.com/eclipse-ee4j/yasson/issues/595
